### PR TITLE
vamp-plugin-sdk 2.6 (new formula)

### PR DESCRIPTION
**Pull request by david0**

### CODE REVIEW

**Comment by dunn** [comment]
A few audit issues:
```
== /usr/local/Library/Formula/vamp-plugin-sdk.rb ==
C:  3: 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  8: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  9: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 10: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 12:  1: Extra blank line detected.
C: 39:  1: Extra empty line detected at class body end.

1 file inspected, 6 offenses detected
Error: 2 problems in 1 formula

==> audit problems
vamp-plugin-sdk:
 * `checksum` (line 6) should be put before `head` (line 4)
 * Non-libraries were installed to "/usr/local/Cellar/vamp-plugin-sdk/2.6/lib"
Installing non-libraries to "lib" is discouraged.
The offending files are:
  /usr/local/Cellar/vamp-plugin-sdk/2.6/lib/libvamp-hostsdk.so.3
  /usr/local/Cellar/vamp-plugin-sdk/2.6/lib/libvamp-hostsdk.so.3.6.0
  /usr/local/Cellar/vamp-plugin-sdk/2.6/lib/libvamp-sdk.so.2
  /usr/local/Cellar/vamp-plugin-sdk/2.6/lib/libvamp-sdk.so.2.6.0
```
For OS X, `libvamp-hostsdk.so.3` should be `libvamp-hostsdk.3.dylib`. If there's no way to configure the build for that, it'd be great if you could report it to the developers.

**Comment by david0** [comment]
I updated to formula, now its green. I also informed upstream about the so vs. dylib naming.

File: Library/Formula/vamp-plugin-sdk.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  depends_on "automake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libsndfile"
+  depends_on "libogg"
+  depends_on "flac"
+
+  # activate osx specific items in Makefile.in
+  # https://code.soundsoftware.ac.uk/issues/1473
+  patch :p0, :DATA
```
Comment (mikemcquaid) [inline Library/Formula/vamp-plugin-sdk.rb]: Thanks for the conversation here but I think I'd rather wait until it reaches a conclusion on an agreed right approach (i.e. a patch that would be merged) before we include this. Thanks, though :heart:

**Comment by david0** [comment]
I integrated the enhancements from upstream and hope its ready to be merged now.

File: Library/Formula/vamp-plugin-sdk.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      const VampPluginDescriptor *
+      vampGetPluginDescriptor(unsigned int version, unsigned int index) { return NULL; }
+    EOS
+
+    system ENV.cxx, "test.cpp", "-I#{include}", "-Wl,-dylib", "-o", "test.dylib"
+    assert_match /Usage:/, shell_output("vamp-rdf-template-generator 2>&1", 2)
+
+    cp "#{lib}/vamp/vamp-example-plugins.so", testpath/"vamp-example-plugins.dylib"
+    ENV["VAMP_PATH"]=testpath
+    assert_match /amplitudefollower/, shell_output("vamp-simple-host -l")
```
Comment (bfontaine) [inline Library/Formula/vamp-plugin-sdk.rb]: Use `#{bin}/vamp-simple-host` here.

**Comment by david0** [comment]
@bfontaine I included the change you suggested, please have a look at it again.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**